### ISSUE TITLE
False positive in 913100 caused by the vega scanner User-Agent fingerprint

### ISSUE THREAD

**Issue by fidoboy** [coreruleset/coreruleset#4776]
### Description

CRS 4.29.0 contains the following entry in `scanners-user-agents.data`:

```text
# https://subgraph.com/vega/
vega
```

Rule `913100` applies `@pmFromFile scanners-user-agents.data` to the `User-Agent` header.

Because `@pmFromFile` performs phrase matching, the entry `vega` also matches when those four characters occur as part of a larger, unrelated token.

I encountered the following legitimate User-Agent:

```text
Mastodon/4.5.11 (http.rb/5.3.1; +https://veganism.social/)
```

Rule `913100` blocks the request with:

```text
ModSecurity: Access denied with code 403 (phase 1).
Matched phrase "vega" at REQUEST_HEADERS:User-Agent.
[msg "Found User-Agent associated with security scanner"]
[data "Matched Data: vega found within REQUEST_HEADERS:User-Agent: Mastodon/4.5.11 (http.rb/5.3.1; +https://veganism.social/)"]
```

The match is caused by the substring:

```text
vega
```

inside:

```text
veganism.social
```

### Problem

The current `vega` entry does not uniquely identify the Vega security scanner. It can match arbitrary User-Agent strings containing the same four-character sequence, for example:

```text
veganer
Lavega
divegara
veganism.social
```

This appears to be an inherent limitation of representing `vega` as an unbounded phrase in `scanners-user-agents.data`.

### Suggested resolution

Could the Vega fingerprint be made more specific so that it identifies the Vega scanner without matching arbitrary occurrences of the string `vega`?

For example, if actual Vega User-Agents have a sufficiently distinctive format, the corresponding fingerprint could be changed to match that format rather than the bare string `vega`.

I am not suggesting that Vega should necessarily be removed from the scanner list. The issue is specifically that `vega` by itself is not an unambiguous User-Agent fingerprint when used with `@pmFromFile`.

### Environment

* OWASP CRS: 4.29.0
* Rule: 913100
* ModSecurity for Apache
* Apache
* Detection phase: 1

Relevant CRS entry:

```text
# https://subgraph.com/vega/
vega
```

Relevant rule:

```apache
SecRule REQUEST_HEADERS:User-Agent "@pmFromFile scanners-user-agents.data" \
    "id:913100,\
    phase:1,\
    block,\
    capture,\
    t:none,\
    msg:'Found User-Agent associated with security scanner',\
    ..."
```

**Comment by airween** [comment]
Hi @fidoboy,

thanks for reporting this.

You're totally right, unfortunately this is a very annoying issue - and recently I faced some similar issues. (Eg. `.boto` pattern occurred a few unpleasant issues)

We definitely need to review these and figure out something.

Next Monday there will be CRS monthly chat, I hope we can discuss this.

**Comment by EsadCetiner** [comment]
I downloaded the tool to get the full User-Agent and this is the one that's configured by default: `Mozilla/4.0 (compatible; MSIE 8.0; Windows NT 5.1; Trident/4.0; .NET CLR 1.1.4322; InfoPath.1; .NET CLR 2.0.50727; .NET CLR 3.0.4506.2152; .NET CLR 3.5.30729; Vega/1.0` However, that User-Agent is not actually being used (Maybe my version is bugged) and is instead setting the User-Agent header to literally `UserAgent`.

Adding a slash to `vega` should fix this false positive.
